File: Advanced/PageAndRDL/CustomPdfExport/VB.NET/PdfRendering/Rendering/PdfConverter.vb
Imports PdfSharp.Drawing
Imports System.Drawing

Friend Class PdfConverter
	Public Const PointsPerInch As Integer = 72
	Public Const TwipsPerPoint As Integer = 20
	Public Const TwipsPerInch As Integer = PointsPerInch * TwipsPerPoint

	Public Shared Function Convert(area As RectangleF) As XRect
		Return New XRect(area.X / TwipsPerPoint, area.Y / TwipsPerPoint, area.Width / TwipsPerPoint, area.Height / TwipsPerPoint)
	End Function

	Public Shared Function Convert(point As PointF) As XPoint
		Return New XPoint(point.X / TwipsPerPoint, point.Y / TwipsPerPoint)
	End Function
End Class

File: Advanced/PageAndRDL/CustomPdfExport/C#/PdfRendering/Rendering/PdfConverter.cs
using System.Drawing;
using PdfSharp.Drawing;

namespace GrapeCity.ActiveReports.Samples.Export.Rendering.PdfSharp
{
	internal static class PdfConverter
	{
		public const int PointsPerInch = 72;
		public const int TwipsPerPoint = 20;
		public const int TwipsPerInch = PointsPerInch * TwipsPerPoint;

		public static XRect Convert(RectangleF area)
		{
			return new XRect(area.X / TwipsPerPoint, area.Y / TwipsPerPoint, area.Width / TwipsPerPoint, area.Height / TwipsPerPoint);
		}

		public static XPoint Convert(PointF point)
		{
			return new XPoint(point.X / TwipsPerPoint, point.Y / TwipsPerPoint);
		}
	}
}
